scripts/nb-tester/test-notebook.py

```diff
@@ -264,12 +260,17 @@ def create_argument_parser() -> argparse.ArgumentParser:
 if __name__ == "__main__":
     args = create_argument_parser().parse_args()
     paths = map(Path, args.filenames or find_notebooks())
-    filtered_paths = filter_paths(paths, args)
+    filtered_paths = list(filter_paths(paths, args))
 
     # Execute notebooks
     start_time = datetime.now()
     print("Executing notebooks:")
-    results = [execute_notebook(path, args) for path in filtered_paths]
+    with concurrent.futures.ThreadPoolExecutor(max_workers=len(filtered_paths)) as executor:
```

Comment:
We want the maximum number of threads since queues are the bottleneck (not CPU).